src/c_types.py

```diff
@@ -652,27 +660,21 @@ def type_to_string(type: CType) -> str:
             return f"{su} {type.type.name}"
         else:
             return f"anon {su}"
-    decl = ca.Decl("", [], [], [], type, None, None)
+    if isinstance(type, FuncDecl) and not name:
+        name = "(*)"
```

Comment:
Yeah, function pointers are weird, given that dereferencing a function pointer yields a function pointer with the same type.

I think we might be able to avoid some of that confusion/complexity if we're working from the asm, because we'll have a bit more control over the ast we build?

I agree revisiting this would be a good idea. I also want to try making specific `Type` representations of functions - maybe the refactor could be incorporated there.